### ISSUE TITLE
Oracle Sequences can't be used in Record.field() after upgrading JOOQ version from 3.10.4 to 3.12.1

### ISSUE THREAD

**Issue by changliwang** [jOOQ/jOOQ#9707]
### Expected behavior and actual behavior:
Before, I used the version of 3.10.4 in my project. We created a util for checking the SQL field is exist or not, and the code as below:
_Record.field("SEQ_USERID"."nextval")_

In the version of 3.10.4, when checking sequence by Record.field("SEQ_USERID"."nextval"), it returns nextval. And then we think this statement can be executed.

But now, we upgraded the version to 3.12.1, and found this code can't work well. 
Record.field("SEQ_USERID"."nextval") returns null, and we defaulted it as an exception.

Is it expected change from the version of 3.10.* to 3.12.*? And could you please share us some suggestions for using the Record.field()?
### Versions:
- jOOQ: 3.12.1
- Java: (Kotlin)

```kotlin
import org.jooq.Record
val reorderedFields = sqlFields.map { name ->
        @Suppress("UNCHECKED_CAST")
        val field = this.field(name) as Field<Any?>? ?:
            throw RuntimeException("Field present in query but not found in record: $name")
        FieldWithValue(field, this.get(field))
    }.toTypedArray()
```

- Database (include vendor): Oracle 11
- OS: 
- JDBC Driver (include name if inofficial driver):

**Comment by changliwang** [comment]
Hi @lukaseder ,
Sorry for incomplete information, please view the complete codes as following:
`fun Record.reorderFields(sql: String): Record {
    val dsl = DSL.using(this.configuration())

    // Try to parse fields from SQL, and return fields name
    val sqlFields = parseSQLFields(sql) ?: return this
    val recordFields = this.fieldsRow().fieldStream()
        .map { it.qualifiedName }
        .collect(Collectors.toList())
    if (sqlFields == recordFields)
        return this

    // Recorder fields and values to match SQL
    val reorderedFields = sqlFields.map { name ->
        @Suppress("UNCHECKED_CAST")
        val field = this.field(name) as Field<Any?>? ?:
            throw RuntimeException("Field present in query but not found in record: $name")
        // record the field's name and value 
        FieldWithValue(field, this.get(field))
    }.toTypedArray()
    return dsl.newRecord(*reorderedFields)
}`

Actually, I extended this method for building a record with field's value. During I try to execute the sql(**the input of reorderFields method**) of `SELECT SEQ_USERID.NEXTVAL FROM DUAL`, the variable of **sqlFields** is `"SEQ_CARDID"."NEXTVAL"`. 
So when I try to get the field by `this.field(name)`, it returns null in 3.12.1. But in 3.10.4, it returns `nextval`.
By the way, the variable of `name` is `"SEQ_CARDID"."NEXTVAL"`!

**Comment by lukaseder** [comment]
Oh, interesting, so the parser is involved. We've improved a lot of things in the parser in the recent releases. It may well be that it wasn't able to parse `sequence_name.nextval` as an expression on a sequence, but thought of it as an ordinary qualified column expression.

I'll validate this ASAP and update this issue with findings. This might be a special case, given the usage of `NEXTVAL`.

**Comment by changliwang** [comment]
Oh, maybe I found this issue's reason. In order to generating keywords in upper case, I used the settings of `withRenderKeywordStyle(RenderKeywordStyle.UPPER)` in my SQL dialects.
And the Context returns SQL as following:
In the version of 3.10, it is `SELECT SEQ_USERID.nextval FROM DUAL`.
But in the version of 3.12, it is `SELECT SEQ_USERID.NEXTVAL FROM DUAL`.

I searched your official API documents, found the `nextval ` is a keyword in the version of 3.12.1, but not in 3.10. And because we defined the sequence field as `DSL.field("nextval", SQLDataType.BIGINT)` by ourselves.
So the method of `Record.reorderFields` returns `SEQ_USERID.NEXTVAL` instead of the defined field of `SELECT SEQ_USERID.nextval`.

I just need to changed the defined field from `DSL.field("nextval", SQLDataType.BIGINT)`  to `DSL.field("NEXTVAL", SQLDataType.BIGINT)` , and then the codes work well.
@lukaseder 
Thanks very much for your response.
